nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java

```diff
@@ -4293,16 +4294,30 @@ private void updateProcessGroup(final ProcessGroup group, final VersionedProcess
         //As Input Port (IP1) originally belonged to PGA the new connection would be incorrectly linked to the old Input Port
         //instead of the one being in PGB, so it needs to be removed first before updating the connections.
 
-        for (final String removedVersionedId : inputPortsRemoved) {
+        Iterator<String> inputPortsRemovedIterator = inputPortsRemoved.iterator();
+        while (inputPortsRemovedIterator.hasNext()) {
+            final String removedVersionedId = inputPortsRemovedIterator.next();
             final Port port = inputPortsByVersionedId.get(removedVersionedId);
             LOG.info("Removing {} from {}", port, group);
-            group.removeInputPort(port);
+            try {
+                group.removeInputPort(port);
+                inputPortsRemovedIterator.remove();
+            } catch (IllegalStateException e) {
+                LOG.info("Removing {} from {} not possible at the moment, connection needs to be updated first.", port, group);
```

Comment:
Even though this is an info log message at first sight my impression was that I as a user need to "update the connections". Please update this to something like "...not possible at the moment, will try again after updating the connections."